app/common/lib/menuUtil.js

```diff
@@ -141,3 +144,40 @@ module.exports.createRecentlyClosedMenuItems = (lastClosedFrames) => {
   }
   return payload
 }
+
+const isItemValid = (currentItem, previousItem) => {
+  if (previousItem && previousItem === CommonMenu.separatorMenuItem) {
+    if (currentItem === CommonMenu.separatorMenuItem) {
+      return false
+    }
+  }
+  return currentItem && (typeof currentItem.label === 'string' || typeof currentItem.type === 'string')
+}
+
+/**
+ * Remove invalid entries from a menu template:
+ * - null or falsey entries
+ * - extra menu separators
+ * - entries which don't have a label or type
+ */
+module.exports.sanitizeTemplateItems = (template) => {
+  const result = template.reduce((previousValue, currentValue, currentIndex, array) => {
+    const result = currentIndex === 1 ? [] : previousValue
+    if (currentIndex === 1) {
+      if (isItemValid(previousValue)) {
+        result.push(previousValue)
+      }
+    }
+    const previousItem = result.length > 0
+      ? result[result.length - 1]
+      : undefined
+    if (isItemValid(currentValue, previousItem)) {
+      result.push(currentValue)
+    }
+    return result
+  })
+
+  return Array.isArray(result)
+    ? result
+    : [result]
+}
```

Comment:
Here is the fix for the bug which led to the revert (which happened with https://github.com/brave/browser-laptop/pull/5867)